subprojects/core/src/main/java/org/gradle/configuration/project/LifecycleProjectEvaluator.java

```diff
@@ -76,7 +77,13 @@ private void notifyAfterEvaluate(ProjectEvaluationListener listener, ProjectInte
         } catch (Exception e) {
             if (state.hasFailure()) {
                 // Just log this failure, and pass the existing failure out in the project state
-                LOGGER.error("Failed to notify ProjectEvaluationListener.afterEvaluate(), but primary configuration failure takes precedence.", e);
+                boolean logStackTraces = project.getGradle().getStartParameter().getShowStacktrace() != ShowStacktrace.INTERNAL_EXCEPTIONS;
+                String infoMessage = "Failed to notify ProjectEvaluationListener.afterEvaluate(), but primary configuration failure takes precedence.";
```

Comment:
I was about to add a comment about the error message. I am bit unsure what we want to express here with "leading to a follow-up error". 

- There are simply two different errors during project configuration: one outside of `afterEvaluate` and one inside.
- Because project evaluation failed the exception in `afterEvaluate` is triggered as well.

As a user I don't think I need to know these implementation details. What I want to know is that there are two issues.